NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -559,47 +601,76 @@ public bool VerifyScript(Script scriptSig, Script scriptPubKey, TransactionCheck
 
 			return true;
 		}
-
-		private bool VerifyWitnessProgram(WitScript witness, WitProgramParameters wit, TransactionChecker checker)
+		const byte ANNEX_TAG = 0x50;
+		private bool VerifyWitnessProgram(WitScript witness, WitProgramParameters wit, TransactionChecker checker, bool isP2SH)
 		{
-			List<byte[]> stack = new List<byte[]>();
+			ContextStack<byte[]> stack = new ContextStack<byte[]>(witness.Pushes);
 			Script scriptPubKey;
-
 			if (wit.Version == 0)
 			{
 				if (wit.Program.Length == 32)
 				{
 					// Version 0 segregated witness program: SHA256(CScript) inside the program, CScript + inputs in witness
-					if (witness.PushCount == 0)
+					if (stack.Count == 0)
 					{
 						return SetError(ScriptError.WitnessProgramEmpty);
 					}
-					scriptPubKey = Script.FromBytesUnsafe(witness.GetUnsafePush(witness.PushCount - 1));
-					for (int i = 0; i < witness.PushCount - 1; i++)
-					{
-						stack.Add(witness.GetUnsafePush(i));
-					}
+					scriptPubKey = Script.FromBytesUnsafe(stack.Pop());
 					var hashScriptPubKey = Hashes.SHA256(scriptPubKey.ToBytes(true));
 					if (!Utils.ArrayEqual(hashScriptPubKey, wit.Program))
 					{
 						return SetError(ScriptError.WitnessProgramMissmatch);
 					}
+					return ExecuteWitnessScript(stack, scriptPubKey, HashVersion.WitnessV0, checker);
 				}
 				else if (wit.Program.Length == 20)
 				{
 					// Special case for pay-to-pubkeyhash; signature + pubkey in witness
-					if (witness.PushCount != 2)
+					if (stack.Count != 2)
 					{
 						return SetError(ScriptError.WitnessProgramMissmatch); // 2 items in witness
 					}
 					scriptPubKey = PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(new KeyId(wit.Program));
-					stack = witness.Pushes.ToList();
+					return ExecuteWitnessScript(stack, scriptPubKey, HashVersion.WitnessV0, checker);
 				}
 				else
 				{
 					return SetError(ScriptError.WitnessProgramWrongLength);
 				}
 			}
+			else if (wit.Version == OpcodeType.OP_1 && wit.Program.Length == 32 && !isP2SH)

```

Comment:
I think this was discussed somewhere else but the `wit.Version` should be `1` instead of `OP_1`

Reply:
On the contrary, it should be OP_1 but not 1